sunpy/image/transform.py

```diff
@@ -167,7 +101,176 @@ def _get_transform_method(method, use_scipy):
+
+            # If missing cannot be used directly, use a value in the input range of the image
+            missing_to_use = (missing if handles_nan_missing or not np.isnan(missing)
+                              else np.nanmin(image))
+
+            if not handles_image_nans:
+                isnan = np.isnan(image)
+                has_nans = np.any(isnan)
+            else:
+                has_nans = False
+
+            # If needed, set any image NaNs to a value that is in the range of the input image
+            image_to_use = np.nan_to_num(image, nan=np.nanmin(image)) if has_nans else image
+
+            rotated_image = rotation_function(image_to_use, matrix, shift, order,
+                                              missing_to_use, clip_to_use)
+
+            # If needed, restore the NaNs
+            if not handles_image_nans and has_nans:
+                # Use a convolution to find all pixels that are affected by NaNs
+                # We want a kernel size that is an odd number that is at least order+1
+                size = 2*int(np.ceil(order/2))+1
+                expanded_nans = convolve2d(isnan.astype(int),
+                                           np.ones((size, size)).astype(int),
+                                           mode='same')
+                rotated_nans = scipy.ndimage.affine_transform(expanded_nans.T, matrix,
+                                                              offset=shift, order=1,
+                                                              mode='nearest').T
+                rotated_image[rotated_nans > 0] = np.nan
+
+            if not handles_nan_missing and np.isnan(missing):
+                # Rotate a constant image to determine where to apply the NaNs for `missing`
+                constant = scipy.ndimage.affine_transform(np.ones_like(image).T.astype(int), matrix,
+                                                          offset=shift, order=0, mode='constant').T
+                rotated_image[constant < 1] = np.nan
+
+            if not handles_clipping and clip and not np.all(np.isnan(rotated_image)):
+                # Clip the image to the input range
+                if np.isnan(missing):
+                    # If `missing` is NaN, clipping to the input range is straightforward
+                    rotated_image.clip(np.nanmin(image), np.nanmax(image), out=rotated_image)
+                else:
+                    # Otherwise, check if `missing` should be considered part of the input range
+                    lower = np.nanmin([np.max([missing, np.nanmin(rotated_image)]), np.nanmin(image)])
+                    upper = np.nanmax([np.min([missing, np.nanmax(rotated_image)]), np.nanmax(image)])
+                    rotated_image.clip(lower, upper, out=rotated_image)
+
+            return rotated_image
+
+        _rotation_function_library[name] = wrapper
+
+        affine_transform.__doc__ += (f"\n    Notes for the ``{name}`` rotation method:"
```

Comment:
Making the sections stand out a little more in the built docs seems like a good idea: https://sunpy--5867.org.readthedocs.build/en/5867/generated/api/sunpy.image.transform.affine_transform.html
```suggestion
        affine_transform.__doc__ += (f"\n    **Notes for the ``{name}`` rotation method:**"
```

Reply:
Fixed